### PR TITLE
Finally write guide on asking for help

### LINKED ISSUE #290: Guidance on how to ask for help
We should have a clear guide on how to get stuck and subsequently ask for help. It should cover how to read (and re-read) workshop instructions (e.g. sometimes reading on past the thing that confused you will help clarify). Also reading error messages (e.g. the important bit is right at the top above the big stack trace. There are line numbers for a reason).

> (No waiting between each step)
> 1. make sure you've understood the error / problem
> 1. google it
> 1. ask another team
> 1. ask a mentor

> Guidance for mentors not to help until you’ve shared what you Googled and tried
> Share your solutions with the group in case others are stuck

### PR DESCRIPTION
**Pull request by oliverjam**
Fix #290

### CODE REVIEW

File: src/course/handbook/asking-for-help.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+It's important to develop your problem-solving and debugging skills. If more experienced people always tell you what's wrong then you'll struggle later in your career. Read the [debugging guide](../debugging/) for more information on solving code problems.
+
+## How to ask
+
+Once you're sure you have a problem you cannot quickly solve on your own you should post a message asking for help in the appropriate channel. This could be your cohort's `#help-and-solutions` Discord channel, or the more general `#facoverflow`, or even somewhere public like Stack Overflow or a project's GitHub Discussions.
+
+### Describe the problem
+
+A good message should describe the problem, and especially what you want to happen. Often people describe a specific syntax issue, when really their problem is with their whole approach. For example for "how can I run these three functions one after the other?" the most helpful solution might be "do it all in a single SQL query instead".
```
Comment (reubengt) [inline src/course/handbook/asking-for-help.md:29]: Maybe we should pick an example more close to the fundamentals so this guide can be used across all stages of FAC?

File: src/course/handbook/asking-for-help.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+```
+````
+
+will produce something like this:
+
+```html
+<h1>Hello world</h1>
+```
+
+If your project is on GitHub then consider linking directly to the lines of code in question. You can do this by [clicking the line number on GitHub](https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/creating-a-permanent-link-to-a-code-snippet). Make sure you have pushed your latest work to a branch so it's up-to-date.
```
Comment (reubengt) [inline src/course/handbook/asking-for-help.md:53]: love this, was a game changer during code reviews in my cohort

File: src/course/handbook/asking-for-help.md (hunk 1 above)
Comment (oliverjam) [reply src/course/handbook/asking-for-help.md:29]: Good point, I was just struggling to think of one. Any better ideas? 😅 

"How can I swap the date and month in this date string?" -> "You shouldn't, you should use the built-in Internationalization API"

File: src/course/handbook/asking-for-help.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
+# Asking for help
+
+Learning how to ask for help effectively is important. Every developer needs help at some point, especially when they're learning new things. You can make it easier for people to help you, and increase the likelihood of solving your problem.
```
[the remark this answers is not in the document]
Comment (oliverjam) [reply src/course/handbook/asking-for-help.md:3]: Yeah I think I over-edited this, will re-work

**Pull request merged**